core/src/main/java/org/apache/accumulo/core/zookeeper/ZcNode.java

```diff
@@ -98,7 +98,6 @@ private ZcNode() {
    * @throws IllegalStateException in the case where the node exists and the data was never set
    */
   byte[] getData() {
-    Preconditions.checkState(cachedData());
```

Comment:
Hopefully this  Preconditions can be added back now.  That check is to defend against calling getData when data was never fetched from zookeeper, in which case a null would returned indicating zoocache knows the node does not exist when that is not true.  Guessing the removal was related to the incorrect mocking of ZK in the unit test.

Reply:
Added this back in db3bf2b